### PR TITLE
[#2525] Check for existence and type when importing project data.

### PR DESCRIPTION
**Pull request by cadecairos**
Lighthouse: https://webmademovies.lighthouseapp.com/projects/65733/tickets/2525

### CODE REVIEW

File: src/core/media.js [hunk 1]
```diff
@@ -14,7 +14,8 @@
     var MEDIA_ELEMENT_SAFETY_POLL_INTERVAL = 500,
         MEDIA_ELEMENT_SAFETY_POLL_ATTEMPTS = 10;
 
-    var __guid = 0;
+    var __guid = 0,
+        __console = window.console;
```
Comment (secretrobotron) [inline src/core/media.js]: I don't think this is entirely necessary, is it?

File: src/core/project.js [hunk 2]
```diff
@@ -5,13 +5,15 @@
 define( [ 'core/eventmanager', 'core/media' ],
         function( EventManager, Media ){
 
+  var __console = window.console,
```
Comment (secretrobotron) [inline src/core/project.js]: Same as https://github.com/mozilla/butter/pull/1236/files#L0R18

File: src/core/project.js [hunk 3]
```diff
@@ -199,10 +201,12 @@ define( [ 'core/eventmanager', 'core/media' ],
             oldTarget.json = targetData;
           }
         }
+      } else if ( __console ) {
+        __console.warn( "Ignored imported target data - Must be in an Array." );
```
Comment (secretrobotron) [inline src/core/project.js]: I think you want to use a period instead of a dash here.

File: src/core/project.js [hunk 4]
```diff
@@ -215,6 +219,8 @@ define( [ 'core/eventmanager', 'core/media' ],
             m.json = mediaData;
           }
         }
+      } else if ( __console ) {
+        __console.warn( "Ignored imported media data - Must be in an Array." );
```
Comment (secretrobotron) [inline src/core/project.js]: Same as https://github.com/mozilla/butter/pull/1236/files#L1R205

File: src/core/track.js [hunk 5]
```diff
@@ -6,6 +6,7 @@ define( [ "./eventmanager", "./trackevent", "./views/track-view" ],
         function( EventManager, TrackEvent, TrackView ){
 
   var __guid = 0,
+      __console = window.console,
```
Comment (secretrobotron) [inline src/core/track.js]: Same as https://github.com/mozilla/butter/pull/1236/files#L1R8

File: src/core/track.js [hunk 6]
```diff
@@ -108,8 +109,12 @@ define( [ "./eventmanager", "./trackevent", "./views/track-view" ],
...
           if( importData.trackEvents ){
             var importTrackEvents = importData.trackEvents;
-            for( var i=0, l=importTrackEvents.length; i<l; ++i ){
-              _this.addTrackEvent( importTrackEvents[ i ] );
+            if ( Array.isArray( importTrackEvents ) ) {
+              for( var i = 0, l = importTrackEvents.length; i < l; ++i ) {
+                _this.addTrackEvent( importTrackEvents[ i ] );
+              }
+            } else if( __console ) {
+              __console.warn( "Ignored imported track event data - must be in an Array." );
```
Comment (secretrobotron) [inline src/core/track.js]: Same as https://github.com/mozilla/butter/pull/1236/files#L1R205

File: src/core/media.js (hunk 1 above)
Comment (cadecairos) [inline src/core/media.js]: in IE9, if I do

`if( window.console) { ... }`

IE throws trying to check window.console for existence.  (which is nuts, imho)

storing a reference like that is the way I'm working around that quirk. thoughts?

Comment (jbuck) [inline src/core/media.js]: Why not take out the warnings?

Comment (cadecairos) [inline src/core/media.js]: @secretrobotron : okay IE9 is trying to make me look crazy. last week it was totally throwing. not anymore. will fix.

@jbuck : We think the warnings will be  helpful when malformed json or json that's not as we expect it is loaded.

also :v:

File: src/core/track.js [hunk 7]
```diff
@@ -110,8 +110,12 @@ define( [ "./eventmanager", "./trackevent", "./views/track-view" ],
           }
           if( importData.trackEvents ){
             var importTrackEvents = importData.trackEvents;
-            for( var i=0, l=importTrackEvents.length; i<l; ++i ){
-              _this.addTrackEvent( importTrackEvents[ i ] );
+            if ( Array.isArray( importTrackEvents ) ) {
+              for( var i = 0, l = importTrackEvents.length; i < l; ++i ) {
+                _this.addTrackEvent( importTrackEvents[ i ] );
+              }
+            } else if( console ) {
```
Comment (secretrobotron) [inline src/core/track.js]: Spacing after `if`.

File: src/core/track.js [hunk 8]
```diff
@@ -110,8 +110,12 @@ define( [ "./eventmanager", "./trackevent", "./views/track-view" ],
           }
           if( importData.trackEvents ){
             var importTrackEvents = importData.trackEvents;
-            for( var i=0, l=importTrackEvents.length; i<l; ++i ){
-              _this.addTrackEvent( importTrackEvents[ i ] );
+            if ( Array.isArray( importTrackEvents ) ) {
+              for( var i = 0, l = importTrackEvents.length; i < l; ++i ) {
```
Comment (secretrobotron) [inline src/core/track.js:114]: One more space after `for`.

File: src/core/media.js [hunk 9]
```diff
@@ -549,11 +549,15 @@
             }
             if( importData.tracks ){
               var importTracks = importData.tracks;
-              for( var i=0, l=importTracks.length; i<l; ++i ){
-                var newTrack = new Track();
-                newTrack.json = importTracks[ i ];
-                _this.addTrack( newTrack );
-                newTrack.updateTrackEvents();
+              if( Array.isArray( importTracks ) ) {
```
Comment (mjschranz) [inline src/core/media.js:574]: Since you're making a point to do some style fixing here, I'll make a point to fix this one space.

Comment (cadecairos) [inline src/core/media.js:574]: none of the other ifs in the file follow that spacing style, so leaving as is.

**Pull request merged**